### PR TITLE
Add memory mangament feature

### PR DESCRIPTION
**Pull request by NickNaso**
Added `MemoryManagement` as class and `AdjustExternalMemory` function as static method.
For more info see the following issue: https://github.com/nodejs/node-addon-api/issues/260

### CODE REVIEW

File: doc/memory_management.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+The `MemoryManagement` class contains functions that give the JavaScript engine
+an indication of the amount of externally allocated memory that is kept alive by 
+JavaScript objects.
+
+## Methods
+
+### AdjustExternalMemory
+
+The function `AdjustExternalMemory` adjusts the amount of registered external
+memory. Used to give the JavaScript engine an indication of the amount of externally
```
Comment (gabrielschulhof) [inline doc/memory_management.md]: "[It is ]~~U~~used ..."

File: doc/memory_management.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+The JavaScript engine uses this to decide when to perform global garbage collections.
+Registering externally allocated memory will trigger global garbage collections
+more often than it would otherwise in an attempt to garbage collect the JavaScript
+objects that keep the externally allocated memory alive.
+
+```cpp
+static int64_t MemoryManagement::AdjustExternalMemory(Env env, int64_t change_in_bytes);
+```
+
+- `[in] env`: The environment in which the API is inoked under.
```
Comment (gabrielschulhof) [inline doc/memory_management.md]: "in[v]oked"

File: test/memory_management.cc [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    int64_t tmp = MemoryManagement::AdjustExternalMemory(info.Env(), kSize);
+    tmp = MemoryManagement::AdjustExternalMemory(info.Env(), -kSize);
+    return Boolean::New(info.Env(), tmp == baseline);
+}
+
+Object InitMemoryManagement(Env env) {
+    Object exports = Object::New(env);
+    exports["externalAllocatedMemory"] = Function::New(env, externalAllocatedMemory);
+    return exports;
+}
```
Comment (gabrielschulhof) [inline test/memory_management.cc]: Please add a newline to the end of the file!

File: test/memory_management.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+const buildType = process.config.target_defaults.default_configuration;
+const assert = require('assert');
+
+test(require(`./build/${buildType}/binding.node`));
+test(require(`./build/${buildType}/binding_noexcept.node`));
+
+function test(binding) { 
+    assert.strictEqual(binding.memory_management.externalAllocatedMemory(), true)
+}
```
Comment (gabrielschulhof) [inline test/memory_management.js]: Ditto.

**Comment by mhdawson** [comment]
Landed as 605aa2babffb58e7b87b816da4324af619dc58e2

**Pull request closed without merging**